### DISCUSSION TITLE
what would be the best way to statusmap a door sensor?

### DISCUSSION THREAD

**Discussion by dwery** [flant/grafana-statusmap#158] [Q&A] [answered]
I have a time series related to a door sensor, with two value only: open and closed.
Only status changes are logged, so from t0 to the first event the status would be unknown. 

Ideally the graph would show every bucket between two status changes in the same color as the preceding status.

**Comment by diafour** [comment, marked as answer]
Hello! You need a feature that can be called "fill the gaps". The statusmap doesn't support this logic right now, it just shows values from a datasource. It is possible to achieve your goal if your time DB supports "filling the gaps". 

Also, check the [natel-discrete-panel](https://grafana.com/grafana/plugins/natel-discrete-panel) that visualize statuses precisely in time with gaps filling.

**Comment by dwery** [comment]
@diafour I'll try that, thanks. I'm using AWS Timestream and, while ti has some interpolation functions, I haven't found an easy way to have it fill the gaps.
